velox/exec/Task.cpp

```diff
@@ -28,6 +28,28 @@
 
 namespace facebook::velox::exec {
 
+namespace {
+void collectSourcePlanNodeIds(
+    const core::PlanNode* planNode,
+    std::unordered_set<core::PlanNodeId>& ids) {
+  if (planNode->sources().empty()) {
+    ids.insert(planNode->id());
```

Comment:
Just curious, if the user repeats source ids in the plan, will that get caught elsewhere?
Else, we could catch here via a `set`.

Reply:
@majetideepak I'm planning to add another check for unique plan node IDs in a follow-up PR.